File: hashmap/isomorphic-strings/isomorphic-strings.ts
function isIsomorphic(s: string, t: string): boolean {
    if (s.length == 1) return true;

    const dict = new Map<string, string>();
    const used = new Set<string>();

    for (let i = 0; i < s.length; i++)
    {
        const a = s[i], b = t[i];

        if (dict.has(a))
        {
            if (dict.get(a)! !== b) return false
            continue;
        }

        if (used.has(b)) return false
        dict.set(a, b);
        used.add(b);
    }

    return true;
};

File: hashmap/isomorphic-strings/isomorphic-strings.cs
public class Solution
{
    public bool IsIsomorphic(string s, string t)
    {
        if (s.Length == 1) return true;

        var dict = new Dictionary<char, char>();
        var used = new HashSet<char>();

        for (int i = 0; i < s.Length; i++)
        {
            char a = s[i], b = t[i];

            if (dict.TryGetValue(a, out var dictedB))
            {
                if (dictedB != b) return false;
                continue;
            }

            if (used.Contains(b)) return false;
            dict[a] = b;
            used.Add(b);
        }

        return true;
    }
}